### DISCUSSION TITLE
Possible to abuse `keep-invalid-txs-in-cache`?

### DISCUSSION THREAD

**Discussion by faustbrian** [tendermint/tendermint#9210] [Q&A] [answered]
So I've been wondering if it would be possible to abuse the `keep-invalid-txs-in-cache` setting to flood a node with invalid transactions that then fill up the cache and steadily increase memory use. For example `mempool.size` could be set to `5000` but then you could in theory spam 10000 invalid transactions that fill up the memory of a node. Is there anything in place that prevents this behaviour or is it advised to not use `keep-invalid-txs-in-cache` if that is a concern? Are there recommended values for `cache_size` to avoid that Tendermint struggles? Are the cached transactions getting pruned at some point?

**Comment by cmwaters** [comment, marked as answer]
There is a configured upper bound on the cache size so invalid transactions also eventually get removed from the cache. Also only the hash of the transaction is saved in the cache not the entire transaction so I think it's very unlikely you will experience a problem with memory.
